xdsl/parser.py

```diff
@@ -2133,25 +2133,24 @@ def to_float(self, parser: Parser) -> float:
             Convert the element to a float value. Raises an error if the type
             is compatible.
             """
-            if not isinstance(self.value, int | float):
-                parser.raise_error("Expected float value", at_position=self.span)
             if self.is_negative:
                 return -float(self.value)
             return float(self.value)
 
         def to_type(self, parser: Parser, type: AnyFloat | IntegerType | IndexType):
-            if isinstance(type, AnyFloat):
-                return self.to_float(parser)
-            elif isinstance(type, IntegerType):
-                return self.to_int(
-                    parser,
-                    type.signedness.data != Signedness.UNSIGNED,
-                    type.width.data == 1,
-                )
-            elif isinstance(type, IndexType):
-                return self.to_int(parser, allow_negative=True, allow_booleans=False)
-            else:
-                assert False, "fatal error in parser"
+            match type:
```

Comment:
I don't think type aliases work for match statements... I extracted it into an if condition, so we can still use the match exhaustiveness on the remaining cases

Reply:
Okay, I looked at the PEP again, and indeed we cannot use pattern matching on anything that is not an object (since we can only pattern match values).
I guess here an `if` actually makes more sense no? It will check for exhaustiveness, as you don't return anything in the default case.